src/class/Socket/Socket.hpp

```diff
@@ -0,0 +1,82 @@
+
+#ifndef SOCKET_HPP
+#define SOCKET_HPP
+
+#include <arpa/inet.h>
+#include <cstring>
+#include <iostream>
+#include <netinet/in.h>
+#include <string>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "../IP/IP.hpp"
+#include "../Port/Port.hpp"
+
+#ifdef TEST_ON
+#define private public
+#define protected public
+#endif
+
+#define MAX_SYS_BACKLOG 128
+
+class Socket
+{
+  public:
+	explicit Socket(IP ip, Port port);
+	Socket(Socket const &src);
+	~Socket();
+
+	IP getIp(void) const;
```

Comment:
when we use getIp() will we need the IP obj or just a string?

Reply:
In fact I think I dont even use it at all, I probably just created the get function because I created a get for everything